copy.go

```diff
@@ -49,12 +49,12 @@ type copyin struct {
 	buffer  []byte
 	rowData chan []byte
 	done    chan bool
-	driver.Result
 
 	closed bool
 
-	sync.Mutex // guards err
+	sync.Mutex // guards err and Result
```

Comment:
whilst you're here, can you make this easier to not make a mistake by going:
```
type copyin struct {
  // ....
  type mu struct {
     sync.Mutex
     driver.Result
     err error
  }
}

```

then use

```
var c copyin
c.mu.Lock() ....
c.mu.Result() ...
```

that way it's clear how the mutex is used.

Reply:
i added the mu struct. by making the err & Result available under a different name, and fixing the uses, i noticed the usage wasn't correct.